ToDoList-Redux/Binders/ReduxFramework.swift

```diff
@@ -14,54 +14,72 @@ import CombineRex
 // MARK: - ACTIONS
 
 enum AppAction {
-    case task(TaskList.Action)
+    case appLifecycle(AppLifecycleAction)
+    case task(TaskAction)
 }
 
+enum TaskAction {
+    case add(String)
+    case remove(IndexSet)
+    case move(IndexSet, Int)
+    case toggle(String)
+}
 
 // MARK: - STATE
 
 struct AppState: Equatable {
     var appLifecycle: AppLifecycle
-    var task: TaskList.State
+    var tasks: [Task]
```

Comment:
I recommend that your AppState tree (and AppAction tree) don't point to View Models / View Events. Usually the View Model is not source-of-truth but a state derived from Model/State, therefore you can easily calculate it out of the State ([Task]). This separation can have benefits in the future, when you need to share the property tasks among different screens and sometimes between different modules. Then your State/Model [Task] is the source-of-truth and the TaskList.State is calculated from that.